framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -539,6 +541,35 @@ protected void postDirectSuperTypes(AnnotatedTypeMirror type,
     }
 
     /**
+     * Gets the type of the resulting constructor call of a MemberReferenceTree.
+     *
+     * @param memberReferenceTree MemberReferenceTree where the member is a constructor
+     * @param constructorType     AnnotatedExecutableType of the declaration of the constructor
+     * @return AnnotatedTypeMirror of the resulting type of the constructor
+     */
+    public AnnotatedTypeMirror getResultingTypeOfConstructorMemberReference(MemberReferenceTree memberReferenceTree,
+                                                                            AnnotatedExecutableType constructorType) {
+        if (memberReferenceTree.getMode() != MemberReferenceTree.ReferenceMode.NEW) {
+            ErrorReporter.errorAbort("");
+        }
+
+        // The return type for constructors should only have explicit annotations from the constructor
+        // Recreate some of the logic from TypeFromTree.visitNewClass here.
+
+        // The return type of the constructor will be the type of the expression of the member reference tree.
+        AnnotatedDeclaredType constructorReturnType = (AnnotatedDeclaredType)
+                fromTypeTree(memberReferenceTree.getQualifierExpression());
+
+        // Keep only explicit annotations and those from @Poly
+        ConstructorReturnUtil.keepOnlyExplicitConstructorAnnotations(this, constructorReturnType, constructorType);
```

Comment:
I like the fatter utility classes because it means few places to look for methods. Also, see comment my comment here: https://github.com/typetools/checker-framework/pull/650#discussion_r59768859